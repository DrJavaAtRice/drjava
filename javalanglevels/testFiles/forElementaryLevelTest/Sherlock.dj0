abstract class AFriend extends Object {
  abstract String getName();
}
class Sherlock extends Object {
  AFriend _friend;
  
  int foo() {
    System.out.println("You're my friend, " + _friend.getName());
    return 5;
  }
}

