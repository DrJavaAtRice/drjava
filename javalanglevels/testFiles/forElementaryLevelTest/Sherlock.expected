// Language Level Converter line number map: dj*->java. Entries: 12
//     1->4         2->5         3->6         4->37        5->38        6->39        7->40        8->41   
//     9->42       10->43       11->44       12->83   
abstract class AFriend extends Object {
  public abstract String getName();

  /** This method is automatically generated by the Language Level Converter. */
  public AFriend() {
    super();
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      AFriend cast = ((AFriend) o);
      return true;
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode();
  }
}
class Sherlock extends Object {
  private final AFriend _friend;
  
  public int foo() {
    System.out.println("You're my friend, " + _friend.getName());
    return 5;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public Sherlock(AFriend _friend) {
    super();
    this._friend = _friend;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public AFriend _friend() {
    return _friend;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        _friend() + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      Sherlock cast = ((Sherlock) o);
      return (_friend() != null && _friend().equals(cast._friend()));
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode() ^ 
        (_friend() == null ? 0 : _friend().hashCode());
  }
}

